src/beta/resources/referral.js

```diff
@@ -0,0 +1,117 @@
+/* eslint-disable no-dupe-class-members,no-unused-vars,class-methods-use-this */
+import superagent from 'superagent';
+
+import User, {propTypes as userPropTypes} from "../../resources/user";
+import RequestError from "../../errors/request";
+
+export default (api) =>
+  // eslint-disable-next-line new-cap
+  class Referral extends User(api) {
+    static _name = 'Referral';
+
+    static _url = 'referral_customers';
+
+    static key = 'user';
+
+    static propTypes = userPropTypes;
+
+    static #stripeCreditCardUrl = 'https://api.stripe.com/v1/tokens';
+
+    /**
+     * List all referrals.
+     * @returns {Promise<never>}
+     */
+    static all() {
+      return this.notImplemented('all');
+    }
+
+    /**
+     * retrieve not implemented
+     * @returns {Promise<never>}
+     */
+    static async retrieve(id, urlPrefix) {
+      return this.notImplemented('retrieve');
+    }
+
+    /**
+     * retrieveMe not implemented
+     * @returns {Promise<never>}
+     */
+    static async retrieveMe() {
+      return this.notImplemented('retrieveMe');
+    }
+
+    /**
+     * updateBrand not implemented
+     * @returns {Promise<never>}
+     */
+    async updateBrand(params) {
+      return this.constructor.notImplemented('updateBrand');
+    }
+
+    /**
+     * Add a credit card to the referral's account.
+     * @param {object} creditCardDetails  - Credit card details:
+     * @param {string} creditCardDetails.number  - Credit card number
+     * @param {string} creditCardDetails.expirationMonth  - Credit card expiration month
+     * @param {string} creditCardDetails.expirationYear  - Credit card expiration year
+     * @param {string} creditCardDetails.cvc  - Credit card CVC
+     * @param {string} primaryOrSecondary  - Whether to add the card as the primary or secondary card (defaults to primary)
+     * @returns {Promise<never>}
+     */
+    async addCreditCard(creditCardDetails = {}, primaryOrSecondary = "primary") {
+      const stripeCreditCardId = await Referral.#sendCardDetailsToStripe(creditCardDetails);
+      await Referral.#sendCardDetailsToEasyPost(stripeCreditCardId, primaryOrSecondary);
+      // nothing to return. If an error wasn't thrown, then the card was added successfully
```

Comment:
This should return a response form EasyPost (EasyPost card object)

Reply:
I'm actually noticing that a lot of things in the Node library don't return as a true class.
For example, we don't actually return a `Brand` object when updating a user's brand; we return the body of the request, which is JSON (which acts like an object because, JSON in JavaScript...), but we're not actually saying "create a Brand object out of this data"

That said, I attempted to do so with a couple things, including `Brand` and here, taking the data that's returned and build a `PaymentMethod` object out of it, but I'm getting complaints about constructors. Which is probably the reason why it is the way it is.

So anyway, this function now returns the body of the final request, which is the JSON representation of a payment method, but isn't technically a `PaymentMethod` object.

Reply:
I ran into this same problem when adding a few rate/brand unit tests when we overhauled the test suite. Can you create an internal task with some details to look into this with the Node lib? We either need to answer why it's doing this or fix it. Good callout.

Reply:
As it stands, it seems to be an issue whenever a class is attempting to return a different type of class other than itself.

For example, most functions in the `User` class return a `User` object, by taking the data from the API response and mapping its properties via [this base `create` function](https://github.com/EasyPost/easypost-node/blob/eb4184d0b33fdf98f90ef56bb6ce59e1c23292f2/src/resources/base.js#L180). But `updateBrand` in the `User` class returns a `Brand` object, which throws the constructor error whenever we try to build a `Brand` object.

I imagine it's due to how the `base` class is designed for object creation, as well as perhaps the module exporting setup.